data/transactions/logic/evalAppTxn_test.go

```diff
@@ -369,8 +369,12 @@ func TestAppAxfer(t *testing.T) {
   itxn_field TypeEnum
   itxn_submit
 `
+
+	was := ep.Proto.LogicSigVersion
+	ep.Proto.LogicSigVersion = 8
 	TestApp(t, "global CurrentApplicationAddress; txn Accounts 1; int 100"+noid+"int 1", ep,
 		fmt.Sprintf("Sender (%s) not opted in to 0", appAddr(888)))
+	ep.Proto.LogicSigVersion = was
```

Comment:
At first, it fell out that inner axfer of 0 got outlawed in v9 (because of the check in the available sets, rather than `availableAsset` which special cased 0). But, I decided to allow it at the availability level, even though it is sure to fail later (not opted in), because it led to more consistency (it's hard to ban using 0 as the XferAsset in an inner, but you can get that without ever using `itxn_field`).

Anyway, I'll clean up and explain the tests.

Reply:
Changed it to test all versions, starting at 5.